### PR TITLE
Remove "onoverconstrained"

### LINKED ISSUE #573: Remove overconstrained event (Why do we have overconstrained event?)
In practice it seems that constraints would never be violated for getUserMedia()-supplied tracks. You get what your device is configured for. In theory, the device might not give you what you ask for, but in practice you should. At least the way Chrome is implemented, you would never reconfigure a camera such that existing tracks become overconstrained.

According to Chrome's GUM implementer @guidou, becoming overconstrained therefor doesn't seem applicable other than in the event of non-device supplied sources e.g. network media sources such as RTCPeerConnection's remote tracks.

In recent discussions (https://github.com/w3c/webrtc-pc/issues/2121) it is pointed out that constraints are under-specified, and it wasn't clear what constraints should be applicable to remote tracks.

So question: _Why do we have overconstrained if not for remote tracks?_

@jan-ivar @alvestrand

### PR DESCRIPTION
**Pull request by armax00**
Fixes #573 

Considering the thread in #573, it seems to me that there is a
relatively large consensus as what the faith of onoverconstrained should
be. This commit attempts to remove onoverconstrained from the
mediastream spec.

### CODE REVIEW

**Comment by alvestrand** [comment]
Always use informative titles.

File: getusermedia.html [hunk 1]
```diff
@@ -119,15 +119,6 @@ <h2>Terminology</h2>
         <p>A <a href="#settings">setting</a> refers to the immediate, current
         value of the source's constrainable properties. Settings are always
         read-only.</p>
-        <p>A source's settings can change dynamically over time due to
-        environmental conditions, sink configurations, or constraint changes. A
-        source's settings must always conform to the current set of
-        <a>required constraints</a> on all attached tracks. A source that cannot
-        conform to these constraints causes affected tracks to become
```
Comment (alvestrand) [inline getusermedia.html]: Don't delete this - it's just reality. Instead change "A source that cannot conform ...." to say "A source may temporarily stray outside the bounds set by constraints, such as when a camera switches to a slower frame rate due to low light conditions. The platform SHOULD try to minimize such excursions as far as possible, but will continue to deliver media even when a temporary condition exists that prevents satisfying the constraint."

Comment (armax00) [reply getusermedia.html]: Thanks, I added the text you suggested (please double-check if I did it right).

**Comment by guidou** [comment]
The section added back talks about sources straying outside constraints, but constraints belong to tracks, not sources.
Maybe say that tracks may temporarily stray outside constraints?

**Comment by armax00** [comment]
> The section added back talks about sources straying outside constraints, but constraints belong to tracks, not sources.
> Maybe say that tracks may temporarily stray outside constraints?

You are right with respect to where the constraints reside, but if I understand correctly the concept of source in the spec is loose and might indicate the devices, for example, that are used to record audio. In that case it would be the devices' conditions that change so that the track cannot satisfy the constraints anymore, hence to me it sounds like we do not need the change. Possibly, a way to rephrase it could be this:

"sources conditions may dynamically change such that the related tracks might not satisfy the constraints any longer"

What do you think?

**Comment by guidou** [comment]
Your proposal sgtm

**Comment by armax00** [comment]
> Your proposal sgtm

Done, a slightly rephrased to match the existing text. PTAL.

File: getusermedia.html [hunk 2]
```diff
@@ -119,15 +119,13 @@ <h2>Terminology</h2>
...
-        <a>overconstrained</a> and therefore <a href="#track-muted">muted</a>.
-        A <a>User Agent</a> attempts to ensure that sources adhere to advanced
-        constraints as closely as possible, see <a href=
-        "#constrainable-interface"></a>.</p>
+        <p>A source conditions may dynamically change, such as when a camera
+        switches to a slower frame rate due to low light conditions. In these
```
Comment (stefhak) [inline getusermedia.html]: s/slower/lower/ ?

File: getusermedia.html [hunk 3]
```diff
@@ -119,15 +119,13 @@ <h2>Terminology</h2>
...
+        cases the tracks related to the affected source might not satisfy the
+        set constraints any longer. The platform SHOULD try to minimize such
+        excursions as far as possible, but will continue to deliver media even
+        when a temporary condition exists that prevents satisfying the
```
Comment (stefhak) [inline getusermedia.html]: I think `temporary`should be removed - there is no way to know whether it's temporary or not usually.

Comment (armax00) [reply getusermedia.html]: I agree with you but I would go to the extent of explicitly writing "temporary or permanently".

**Comment by armax00** [comment]
Please, take a look :)

**Comment by armax00** [comment]
Friendly ping.

**Comment by alvestrand** [comment]
Seems ready to merge, but let's check with @dontcallmedom for completeness before merging.

**Comment by dontcallmedom** [comment]
LGTM as well except indeed for the  stray `</p>`

**Comment by armax00** [comment]
> Lgtm modulo stray </p>

What I do not understand is why that </p> should be spurious. There is a <p> open above that would otherwise never get closed. Maybe it should be placed before <ul> ?

**Comment by armax00** [comment]
After two tests we did with hbos, it seems that Travis has something wrong in how it is creating the checkout because is always failing with the same hash (which makes me believe it is not the specific pull request but rather something wrong from Travis' side).

**Comment by henbos** [comment]
Travis is broken, it's not this PR. I got it here too #580:

> Unhandled rejection Error: Command failed: /usr/bin/git checkout 7b144b3b83ce296d5b29c5e4fc2ea2a65e9d3f32
> fatal: reference is not a tree: 7b144b3b83ce296d5b29c5e4fc2ea2a65e9d3f32

Exact same problem

**Pull request merged**